### DISCUSSION TITLE
why themes folder is empty? 为何 themes 文件夹下没有文件？

### DISCUSSION THREAD

**Discussion by johnbao6** [hexojs/hexo#5449] [Question / Answer] [answered]
### Check List

- [ ] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting).
- [X] I have already searched existing issues and they are not help to me.
- [ ] I examined error or warning messages and it's difficult to solve.
- [ ] I am using the [latest](https://github.com/hexojs/hexo/releases) version of Hexo. (run `hexo version` to check)
- [ ] My Node.js is matched [the required version](https://hexo.io/docs/#Required-Node-js-version).

### Expected behavior

The files and folders appear to be in the wrong location.

文件和文件夹位置似乎不对。

### Actual behavior

After installing Hexo, run the command: **hexo init myhexo**, directory as following:
![Screenshot from 2024-03-31 12-12-17](https://github.com/hexojs/hexo/assets/24749194/f5bb3a8e-dc43-4d71-9489-45afce54c01b)

It is strange is that file _config.landscape.yml appears, and the folder **themes** is _empty_, but there is a folder **hexo-theme-landscape** in the folder **node_modules**, which should be the default theme of hexo.

I try to install a new theme: **npm i hexo-theme-stellar**, the directory structure is still as shown above, the folder **themes** is still empty, and a new folder **hexo-theme-stellar** appears in the folder **node_modules**.

I think hexo-theme-landscape and hexo-theme-stellar should be inside the folder themes, and the location of the **_config.landscape.yml** file is also wrong.

Why？Does anyone else have similar issues?

安装好 Hexo 后，运行命令: hexo init myhexo， 查看目录如上图。

有点奇怪是出现了文件 _config.landscape.yml，而且文件夹 themes 是空的，而在 文件夹 node_modules 内有文件夹 hexo-theme-landscape，这应该是 hexo 缺省的主题。

如果试图安装新的主题， npm i hexo-theme-stellar，那么目录结构还是如上图，文件夹 themes 还是空的，而在文件夹 node_modules 内出现了新的文件夹 hexo-theme-stellar。

我认为 hexo-theme-landscape 和 hexo-theme-stellar 应该在文件夹 themes 里面才对，而且 _config.landscape.yml 文件的位置也不对。

为和会这样？其他人也有类似问题吗？

### How to reproduce?

1. hexo init myhexo
2. ll
3. npm i hexo-theme-stellar
4. ll

### Is the problem still there under `Safe mode`?

yes

### Your Node.js & npm version

```text
v21.7.1
10.5.0
```

### Your Hexo and Plugin version

```text
hexo-site@0.0.0 /home/bao/myhexo
├── hexo-generator-archive@2.0.0
├── hexo-generator-category@2.0.0
├── hexo-generator-index@3.0.0
├── hexo-generator-tag@2.0.0
├── hexo-renderer-ejs@2.0.0
├── hexo-renderer-marked@6.2.0
├── hexo-renderer-stylus@3.0.1
├── hexo-server@3.0.0
├── hexo-theme-landscape@1.0.0
├── hexo-theme-stellar@1.27.0
└── hexo@7.1.1
```

### Your `package.json`

```json
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "hexo generate",
    "clean": "hexo clean",
    "deploy": "hexo deploy",
    "server": "hexo server"
  },
  "hexo": {
    "version": ""
  },
  "dependencies": {
    "hexo": "^7.0.0",
    "hexo-generator-archive": "^2.0.0",
    "hexo-generator-category": "^2.0.0",
    "hexo-generator-index": "^3.0.0",
    "hexo-generator-tag": "^2.0.0",
    "hexo-renderer-ejs": "^2.0.0",
    "hexo-renderer-marked": "^6.0.0",
    "hexo-renderer-stylus": "^3.0.0",
    "hexo-server": "^3.0.0",
    "hexo-theme-landscape": "^1.0.0",
    "hexo-theme-stellar": "^1.27.0"
  }
}
```

**Comment by stevenjoezhang** [comment, marked as answer]
Hexo现已支持通过npm安装主题，主题不会出现在themes目录下。如果您需要主题出现在themes目录下，可以使用git clone等其他方式进行安装。

**Reply by johnbao6** [reply]
就是说我说的不是什么问题？包括初始命令 hexo init myhexo 产生的  _config.landscape.yml 的位置也是如此？

这让我感到很奇怪，然而我看到的教程中都没有提到这个问题（包括官网），与其这样，还不如删了目录 themes。
这可能不算 bug，但是很容易让人误解，考虑到 Hexo 的极大影响力，这个问题有些不应该。

**Reply by stevenjoezhang** [reply]
如果主题不支持通过 npm 进行安装，那么主题文档可能会指示用户将其代码仓库 git clone 到 themes 目录下，故该目录得到了保留。不同主题可能根据是否给用户提供足够大的自定义空间而推荐不同的安装方式，因此建议参照主题的文档进行安装，Hexo 文档仅做出了一般的描述。
Hexo 文档确实没有强调 `node_modules` 也是主题的合法位置，相关文档有待进一步完善（https://github.com/hexojs/site/issues/1446 ）。但我个人并不认为这一问题容易令人误解。Hexo 自 5.0 版本起支持通过 npm 安装主题，这一版本发布于 2020 年 7 月。我们在过去并没有收到关于主题安装和配置方式的太多提问，因此相信这一使用方式已经得到了验证。

**Comment by uiolee** [comment]
These are two installation methods. 

- installing through npm,
- installing to `themes/` manually.

**Comment by D-Sketon** [comment]
_config.landscape.yml 的位置官网有明确说明 https://hexo.io/zh-cn/docs/configuration#使用代替主题配置文件
> 独立的主题配置文件应放置于**站点根目录下**，支持 yml 或 json 格式。需要配置站点 _config.yml 文件中的 theme 以供 Hexo 寻找 _config.[theme].yml 文件。
